### PR TITLE
migrate contract registration to curl

### PR DESCRIPTION
**Pull request by philipstaffordwood**
this moves from using the embedded contract in a postman test to using the security contract in the `doc/` folder

### CODE REVIEW

**Comment by philipstaffordwood** [comment]
*NOTE:*
There might be required contract fixes after this PR.

After 
```
docker-compose up
...
```

I can get a valid looking JWT so the contact application seems correct:

```bash
TOKEN=`curl \
-s -v \
-X POST http://localhost:80/connect/token \
-H 'Content-Type: application/x-www-form-urlencoded' \
-H 'cache-control: no-cache' \
-d 'grant_type=password&username=dokuti-admin&password=Password1#&client_id=dokuti-test-client&client_secret=secret&scope=dokuti' \
| jq '.access_token' -r` \
&& echo "TOKEN is :$TOKEN"
```

```JSON
{
  "nbf": 1578475296,
  "exp": 1578478896,
  "iss": "http://localhost",
  "aud": "dokuti",
  "client_id": "dokuti-test-client",
  "sub": "ac3692e7-fa92-43eb-aa35-f1e7f26a4ccf",
  "auth_time": 1578475296,
  "idp": "local",
  "permission": [
    "dokuti.attributes.create",
    "dokuti.attributes.delete",
    "dokuti.attributes.read",
    "dokuti.attributes.update",
    "dokuti.documents.create",
    "dokuti.documents.read",
    "dokuti.documents.update"
  ],
  "team": [
    "fa068cbe-8407-4e83-94af-c69d51f5107b",
    "fe401700-fb64-4717-9ec7-49592962fe64"
  ],
  "email": "dokuti-admin@localhost",
  "username": "dokuti-admin",
  "scope": [
    "dokuti"
  ],
  "amr": [
    "pwd"
  ]
}
```

But document creation tests fail with missing permissions

File: quickstart/docker-compose/docker-compose.yml [hunk 1]
```diff
@@ -12,16 +12,17 @@ services:
...
     ports:
      - "8181:8080"
     networks:
       - dokuti-quickstart       
     volumes:
       - ./sample-application.yaml:/config/application.yaml
       - ./files:/tmp/dokuti-local-uploads
+      - ../../docker-container-scripts/wait-for.sh:/wait-for.sh
+      - ../../docker-container-scripts/entrypoint.sh:/entrypoint.sh
     depends_on:
```
Comment (melissapalmer) [inline quickstart/docker-compose/docker-compose.yml]: These 2 lines are not needed are they? Those are put onto the Docker image for dokuti at: 
- https://github.com/GrindrodBank/dokuti/blob/055d39a1420978a6748f6943868abe98bd3e8010/Dockerfile#L16

File: quickstart/docker-compose/docker-compose.yml [hunk 2]
```diff
@@ -12,16 +12,17 @@ services:
     ## To persist database
     # volumes:
     #   - 'postgresql_data:/bitnami'
-  web:
-    build: 
-      context: ../../
+  dokuti:
```
Comment (melissapalmer) [inline quickstart/docker-compose/docker-compose.yml]: Probably worth updating the base readme as it mentions: 
`Web 1 - The actual document service installed on port 8181`

File: quickstart/docker-compose/docker-compose.yml (hunk 1 above)
Comment (philipstaffordwood) [reply quickstart/docker-compose/docker-compose.yml]: Thx. fixed. 
I didn't `docker pull grindrodbank/dokuti:latest`, my `latest` didn't have it in.

File: quickstart/docker-compose/docker-compose.yml (hunk 2 above)
Comment (philipstaffordwood) [reply quickstart/docker-compose/docker-compose.yml]: well caught.
Fixed

**Comment by melissapalmer** [comment]
> _NOTE:_
> There might be required contract fixes after this PR.
> 
> After
> 
> ```
> docker-compose up
> ```
> 
> I can get a valid looking JWT so the contact application seems correct:
> 
> ```shell
> TOKEN=`curl \
> -s -v \
> -X POST http://localhost:80/connect/token \
> -H 'Content-Type: application/x-www-form-urlencoded' \
> -H 'cache-control: no-cache' \
> -d 'grant_type=password&username=dokuti-admin&password=Password1#&client_id=dokuti-test-client&client_secret=secret&scope=dokuti' \
> | jq '.access_token' -r` \
> && echo "TOKEN is :$TOKEN"
> ```
> 
> Using http://jwt.io to decode:
> 
> ```json
> {
>   "nbf": 1578475296,
>   "exp": 1578478896,
>   "iss": "http://localhost",
>   "aud": "dokuti",
>   "client_id": "dokuti-test-client",
>   "sub": "ac3692e7-fa92-43eb-aa35-f1e7f26a4ccf",
>   "auth_time": 1578475296,
>   "idp": "local",
>   "permission": [
>     "dokuti.attributes.create",
>     "dokuti.attributes.delete",
>     "dokuti.attributes.read",
>     "dokuti.attributes.update",
>     "dokuti.documents.create",
>     "dokuti.documents.read",
>     "dokuti.documents.update"
>   ],
>   "team": [
>     "fa068cbe-8407-4e83-94af-c69d51f5107b",
>     "fe401700-fb64-4717-9ec7-49592962fe64"
>   ],
>   "email": "dokuti-admin@localhost",
>   "username": "dokuti-admin",
>   "scope": [
>     "dokuti"
>   ],
>   "amr": [
>     "pwd"
>   ]
> }
> ```
> 
> But document creation tests fail with what looks like missing permissions.

Is it possible that the latest contract is actually in the postman collection and that needs to be bought down into the file? 

PS: can we also remove the postman collection with security contract from with this PR?

**Pull request merged**